mlflow/server/js/src/components/ExperimentPage.js

```diff
@@ -23,42 +25,61 @@ class ExperimentPage extends Component {
     dispatchSearchRuns: PropTypes.func.isRequired,
   };
 
-  state = {
-    paramKeyFilter: new KeyFilter(),
-    metricKeyFilter: new KeyFilter(),
+  static defaultState = {
+    paramKeyFilterString: "",
+    metricKeyFilterString: "",
     getExperimentRequestId: getUUID(),
     searchRunsRequestId: getUUID(),
     searchInput: '',
     lastExperimentId: undefined,
     lifecycleFilter: LIFECYCLE_FILTER.ACTIVE,
   };
 
+  static getStore(experimentId) {
+    return LocalStorageUtils.getStore("ExperimentPage", experimentId);
+  }
+
+  store = ExperimentPage.getStore(this.props.experimentId);
+  state = LocalStorageUtils.loadComponentState(this.store, ExperimentPage.defaultState);
+
+  /** Wrapper over setState that caches certain fields in local storage. */
+  setStateWrapper(newState) {
+    const { paramKeyFilterString, metricKeyFilterString, searchInput } = newState;
+    this.setState(newState, () => {
+      LocalStorageUtils.saveComponentState(this.store, {
+        paramKeyFilterString,
+        metricKeyFilterString,
+        searchInput,
+      });
+    });
+  }
+
   static getDerivedStateFromProps(props, state) {
     if (props.experimentId !== state.lastExperimentId) {
+      const store = ExperimentPage.getStore(props.experimentId);
       const newState = {
-        paramKeyFilter: new KeyFilter(),
-        metricKeyFilter: new KeyFilter(),
+        ...LocalStorageUtils.loadComponentState(store, ExperimentPage.defaultState),
```

Comment:
This isn't my favorite API, as the set of keys passed to newState is not clear. I think it would be slightly preferable if we could define an explicit class-like object which had the fields we wanted (e.g., `SearchState`) or if we desugared the state and passed the individual parameters directly in.